frontend/packages/dev-console/src/components/topology/components/nodes/PodSet.tsx

```diff
@@ -67,8 +69,18 @@ const PodSet: React.FC<PodSetProps> = ({ size, data, x = 0, y = 0, showPodCount
     data.isRollingOut,
   );
 
+  const [hpa] = useRelatedHPA(
+    data.dc.apiVersion,
+    data.dc.kind,
+    data.dc.metadata.name,
+    data.dc.metadata.namespace,
+  );
+  const hpaControlledScaling = !!hpa;
+
   const obj = get(data, ['current', 'obj'], null) || data.dc;
-  const { title, subTitle, titleComponent } = podRingLabel(obj, data.dc.kind, data?.pods);
+  const { title, subTitle, titleComponent } = hpaControlledScaling
+    ? hpaPodRingLabel(obj, hpa, data?.pods)
```

Comment:
Why bother reading from the HPA? The workload should continue to have the spec and status needed to populate the pod ring.